tools/validate/validate.cpp

```diff
@@ -68,26 +68,20 @@ void scan_file(const filesystem::path& filepath, const TabPolicy tab_policy, vec
     size_t tab_characters            = 0;
     size_t trailing_whitespace_lines = 0;
 
+    constexpr size_t MaxErrorsForErrorLinesReported = 10; // per each error type we report
+
+    array<size_t, MaxErrorsForErrorLinesReported> overlength_line_occurecnes;

```

Comment:
```suggestion
    array<size_t, MaxErrorsForErrorLinesReported> lines_over_length_restriction{};
```

I also really don't understand why we're putting this here as opposed to printing as they come up...

Reply:
We first check for overlength lines, then if found, check for extension:
https://github.com/microsoft/STL/blob/314f65f831f827dbfb05ebcc0f08c496f5ddf64f/tools/validate/validate.cpp#L166-L185

Then finally print.

So in best case we avoid checking extension (a bit expensive operation), still with overlength lines we avoid printing for mismatched extension (must avoid).

Printing as we go with still avoiding unnecessary extension check will make us checking on the first overlength line. Which is probably fine, but makes this change more complicated than it was originally intended.